include/npcomp/Dialect/Refback/IR/RefbackOps.td

```diff
@@ -37,4 +37,38 @@ def Refback_AllocMemRefOp : Refback_Op<"alloc_memref", []> {
   let assemblyFormat = "$shape attr-dict `:`  type($memref)";
 }
 
+//===----------------------------------------------------------------------===//
+// Ops related to Torch.
+//===----------------------------------------------------------------------===//
+
+def Refback_TorchFallbackOp : Refback_Op<"torch_fallback",
+                           [SingleBlockImplicitTerminator<"TorchFallbackYieldOp">,
+                            DeclareOpInterfaceMethods<RegionBranchOpInterface>,
+                            RecursiveSideEffects]> {
+  let summary = "Execute region through the Torch fallback.";
+  let description = [{
+    Execute region through the Torch fallback.
+  }];
+  let arguments = (ins Variadic<AnyType>:$operands);
+  let regions = (region SizedRegion<1>:$doRegion);
+  let results = (outs Variadic<AnyType>:$results);
+  let printer = [{ return ::print(p, *this); }];
+  let parser = [{ return ::parse$cppClass(parser, result); }];
+  let hasCanonicalizer = 1;
+}
+
+def Refback_TorchFallbackYieldOp : Refback_Op<"torch_fallback_yield",
+       [NoSideEffect, ReturnLike, Terminator, HasParent<"TorchFallbackOp">]> {
+  let summary = "Yield operation";
+  let description = [{
+    This yield operation represents a return operation within the
+    `refback.torch_fallback` operation region. The operation takes variable number of
+    operands and produces no results. The operand number and types must match
+    the number and types of parent `refback.torch_fallback` results.
+  }];
+  let arguments = (ins Variadic<AnyType>:$operands);
```

Comment:
just tensors for now

Reply:
Currently discussing switching to a tensor-only I/O with Bryce Arden. If I understand correctly, NumPy arrays need to be lowered to tensors. Is there already code that lowers NumPy arrays?

Reply:
Started adding one here that is enough for recognized ops in simple cases: https://github.com/llvm/mlir-npcomp/pull/200

Hard part here is that we are dealing with a fallback kernel, so we don't in general know all its aliasing properties. Refbackend is currently tensor-only (does not accept arrays). For now try to fallback only for sigIsMutable=false ops, and insert numpy.copy_to_tensor/numpy.create_array_from_tensor around the op. Combined with numpy-public-functions-to-tensor + numpy-array-to-tensor should be enough for now.